magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleRegistry.kt

```diff
@@ -8,7 +8,7 @@ import com.wealthfront.magellan.lifecycle.LifecycleState.Shown
 
 internal class LifecycleRegistry : LifecycleAware {
 
-  var listeners: List<LifecycleAware> = emptyList()
+  var listeners: Set<LifecycleAware> = emptySet()
```

Comment:
Hmm, I think this changes the ordering of lifecycle event propagation, which could be very bad. I think TreeSet preserves ordering?

Reply:
fixed. I think TreeSet sorts the items. I need a LinkedHashSet